### PR TITLE
examples: add new examples to send and receive User Data Relay Frames

### PR DESCRIPTION
**Pull request by hgonzaleDigi**
- Included generic examples for sending and receiving data.
- Included specific interface examples for Bluetooth and MicroPython for
  sending and receiving data over the specific interface.
- Included a Bluetooth example to receive an entire file.

Signed-off-by: Hector Gonzalez <hector.gonzalez@digi.com>

### CODE REVIEW

File: examples/communication/relay/ReceiveUserDataRelaySample/ReadMe.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+  Introduction
+  ------------
+  This sample C# application shows how to receive and process data coming
+  from the bluetooth interface of the XBee device using a callback executed
```
Comment (rubenmoral) [inline examples/communication/relay/ReceiveUserDataRelaySample/ReadMe.txt]: I would say "from other interfaces of the XBee device", the example is generic for any interface.

**Review by rubenmoral — APPROVED** [review]

File: examples/communication/relay/ReceiveUserDataRelaySample/ReadMe.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    2) Wait until your XBee device is discovered and connect to it with the
+       password you configured during the setup.
+
+    3) Open the Relay Console from the Options menu.
+
+    4) Tap on the Add (+) button to create a new frame. Select 'SERIAL' as
+       Relay interface and enter 'Hello XBee!' as data. Then, tap on 'Add'.
+
+    5) Send this frame by selecting it and taping on 'Send selected frame'.
```
Comment (diescalo) [inline examples/communication/relay/ReceiveUserDataRelaySample/ReadMe.txt]: Typo: ``tapping``

File: examples/communication/bluetooth/ReceiveBluetoothDataSample/ReadMe.txt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    2) Wait until your XBee device is discovered and connect to it with the
+       password you configured during the setup.
+
+    3) Open the Relay Console from the Options menu.
+
+    4) Tap on the Add (+) button to create a new frame. Select 'SERIAL' as
+       Relay interface and enter 'Hello XBee!' as data. Then, tap on 'Add'.
+
+    5) Send this frame by selecting it and taping on 'Send selected frame'.
```
Comment (diescalo) [inline examples/communication/bluetooth/ReceiveBluetoothDataSample/ReadMe.txt]: Typo: ``tapping``

File: examples/communication/micropython/SendMicroPythonDataSample/ReadMe.txt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  NOTE: This example uses the generic XBee device (XBeeDevice) class,
+        but it can be applied to any other local XBee device class.
+
+
+  Files
+  -----
+    * MainApp.cs:
+      Main application class. It instantiates an XBee device, establishes a
+      serial connection with it and sends the data to the XBee MicroPython
+      interface. Finally, it prints out the result of the sent operation.
```
Comment (diescalo) [inline examples/communication/micropython/SendMicroPythonDataSample/ReadMe.txt]: Shouldn't it be ``send operation``?

File: examples/communication/relay/SendUserDataRelaySample/ReadMe.txt [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  NOTE: This example uses the generic XBee device (XBeeDevice) class,
+        but it can be applied to any other local XBee device class.
+
+
+  Files
+  -----
+    * MainApp.cs:
+      Main application class. It instantiates an XBee device, establishes a
+      serial connection with it and sends the data to the specified XBee
+      interface. Finally, it prints out the result of the sent operation.
```
Comment (diescalo) [inline examples/communication/relay/SendUserDataRelaySample/ReadMe.txt]: Shouldn't it be send operation?

**Review by diescalo — APPROVED** [review]

**Pull request merged**